doc/introduction/interfaces.rst

```diff
@@ -3,17 +3,99 @@
+that QNode is requested.
+
+.. code-block:: python
+
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.probs(wires=0)
+
+PennyLane currently provides the following differentiation methods for QNodes, divided into
+the following three categories.
+
+.. note::
+
+    If not specified, the default differentiation method is ``diff_method="best"``. PennyLane
+    will attempt to determine the *best* differentiation method given the device and interface.
+    Typically, PennyLane will prioritize device-provided gradients, backpropagation, parameter-shift
+    rule, and finally finite-differences, in that order.
+
+Device gradients
+~~~~~~~~~~~~~~~~
+
+* ``"device"``: Queries the device directly for the gradient.
+  Only allowed on devices that provide their own gradient computation.
+
+Reverse-mode differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods only support state-vector simulators, they are **not** hardware compatible;
+they are only supported on *statevector* simulator devices such as :class:`default.qubit
+<~.DefaultQubit>`.
+
+However, for rapid prototyping on simulators, these methods typically out-perform forward-mode
+accumulators such as the parameter-shift rule and finite-differences. For more details, see the
+:doc:`quantum backpropagation <demos/tutorial_backprop>` demonstration.
+
+* ``"backprop"``: Use standard backpropagation.
+
+  This differentiation method is only allowed on simulator
+  devices that are classically end-to-end differentiable, for example
+  :class:`default.qubit <~.DefaultQubit>`. This method does *not* work on devices
+  that estimate measurement statistics using a finite number of shots; please use
+  the ``parameter-shift`` rule instead.
+
+* ``"adjoint"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  The `adjoint method <https://arxiv.org/abs/2009.02823>`__  reverses through the circuit after a
+  forward pass by iteratively applying the inverse (adjoint) gate. This method is similar to
+  ``"backprop"``, but trades off increased runtime with significantly lower memory usage.
+
+* ``"reversible"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  This method is similar to the ``adjoint`` method, but has a slightly larger time overhead and a similar
+  memory overhead.
+
+  Compared to the parameter-shift rule, the reversible method can be faster or slower, depending on
+  the density and location of parametrized gates in a circuit.
+
+Forward-mode differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods support both quantum hardware and simulators. However, when using a simulator,
+you may notice that the time required to compute the gradients scales
+with the number of trainable circuit parameters.
```

Comment:
It's also true that the time scales with number of params for methods like adjoint. Could we just say that it may be slower on a simulator to use forward mode?

Reply:
Wait really? I didn't know that. I thought the adjoint method was a reverse accumulator, in that it scales with the dimension of the output, not the input of the function being differentiated?

Reply:
Oh, there is a slightly subtlety here I think?

for a function R^m -> R^n, 

* forward accumulators (finite diff, param shift) require `m` quantum evaluations.

* Reverse accumulators instead require 'n' quantum evaluations (this can be seen somewhat from the line

  ```python
  lambdas = [self._apply_operation(lambda_, op) for lambda_ in lambdas]
  ```

  in the adjoint logic, where `len(lambda) ~ tape.output_dim`).

But the reverse accumulators will still somewhat scale with the number of parameters insomuch as more parameters = deeper circuits? And in the adjoint case, more generators to compute?

Reply:
Changed this sentence to 'scales quadratically' to be more specific.